app/components/FilterPanel.tsx

```diff
@@ -0,0 +1,204 @@
+// app/addform/components/FilterPanel.tsx
+'use client';
+import React from 'react';
+import type { Offer } from '../data/offersElectronic';
+
+export type Filters = {
+    minPrice?: number | null;
+    maxPrice?: number | null;
+    state?: 'all' | 'new' | 'used';
+    location?: string;
+    brand?: string;
+    keyword?: string;
+
+    //how long - time ago was published
+    daysAgo?: number | 'all';
+};
+
+type Props = {
+    allOffers: Offer[];
+    filters: Filters;
+    setFilters: (f: Filters) => void;
+};
+
+export default function FilterPanel({ allOffers, filters, setFilters }: Props) {
+    // Extract unique brands and locations for filter dropdowns
+    const brands = Array.from(new Set(allOffers.map((o) => o.brand))).sort();
+    const locations = Array.from(
+        new Set(allOffers.map((o) => o.location))
+    ).sort();
+
+    return (
+        <aside className='mb-6 rounded border bg-white p-4 shadow-sm'>
+            <h3 className='mb-2 font-semibold'>Filtry (Elektronika)</h3>
+
+            <div className='grid gap-3 sm:grid-cols-2'>
+                <div>
+                    <label className='block text-xs text-gray-600'>
+                        Szukaj (tytuł)
+                    </label>
+                    <input
+                        value={filters.keyword ?? ''}
+                        onChange={(e) =>
```

Comment:
please wrap it with separate function - it should look like this
```suggestion
                        onChange={updateFilters}
```